src/box/iproto.cc

```diff
@@ -3397,6 +3490,26 @@ iproto_do_cfg_f(struct cbus_call_msg *m)
 		iproto_thread_accept(iproto_thread, io, addr, addrlen, session);
 		break;
 	}
+	case IPROTO_CFG_DROP_CONNECTIONS: {
+		struct iproto_connection *con;
+		static const struct cmsg_hop cancel_route[1] =
+						{{ tx_process_cancel, NULL }};
+		rlist_foreach_entry(con, &iproto_thread->connections,
+				    in_connections) {
+			if (!con->is_in_replication)
+				iproto_connection_close(con);
```

Comment:
Would be nice to have a comment somewhere explaining why we skip replication connections.

Reply:
Done.

```diff
@@ -3486,6 +3486,12 @@ iproto_do_cfg_f(struct cbus_call_msg *m)
                                {{ tx_process_cancel_inprogress, NULL }};
                rlist_foreach_entry(con, &iproto_thread->connections,
                                    in_connections) {
+                       /*
+                        * Replication IO is done outside iproto so we
+                        * cannot close them as usual. Anyway we cancel
+                        * replication fibers as well and close connection
+                        * after replication is breaked.
+                        */
                        if (!con->is_in_replication &&
                            con->state == IPROTO_CONNECTION_ALIVE)
                                iproto_connection_close(con);
```